packages/next/src/lib/metadata/default-metadata.tsx

```diff
@@ -1,12 +1,17 @@
 import React from 'react'
 import type { ResolvedMetadata } from './types/metadata-interface'
 
-export const createDefaultMetadata = (): ResolvedMetadata => {
+export function createDefaultMetadata(): ResolvedMetadata {
+  const defaultMetadataBase =
+    process.env.NODE_ENV === 'production' && process.env.VERCEL_URL
+      ? new URL(`https://${process.env.VERCEL_URL}`)
```

Comment:
Note that `VERCEL_URL` will not be the domain visited but will be the unique url. Are you sure you want to expose this by default? It could allow someone to use a pinned version of your application instead of the intended domain that is always latest.

Reply:
yes, especially when you're developing a og card, or tweaking around other metadata assets, you want to see if that preview url is working based on your changes instead of always requesting the prod major version